### PR TITLE
Use separate context for Jira webhook retry

### PR DESCRIPTION
**Pull request by EdwardDowling**
Part of #34581 

Prevent context used for retrying reconciliation of webhook and http client results.

### CODE REVIEW

File: integrations/access/jira/app.go [hunk 1]
```diff
@@ -290,6 +290,9 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 
 // onJiraWebhook processes Jira webhook and updates the status of an issue
 func (a *App) onJiraWebhook(ctx context.Context, webhook Webhook) error {
+	ctx, cancel := context.WithTimeout(ctx, webhookIssueAPIRetryTimeout)
```
Comment (r0mant) [inline integrations/access/jira/app.go:293]: Looks like webhookIssueAPIRetryTimeout is 1 minute, should be bump it as well to, say, 5?

Comment (zmb3) [inline integrations/access/jira/app.go:293]: This context will still be canceled when the parent context is canceled, is that what you intend?

Comment (r0mant) [reply integrations/access/jira/app.go:293]: Ah yes, Zac is right, I don't think the fix will work right now. We need a separate context here.

Comment (EdwardDowling) [reply integrations/access/jira/app.go:293]: I think that should be fine if that only happens when the cancel func of the parent or child is called. As long as this updates the timeout correctly I think[ it should be fine](https://go.dev/play/p/Oz4d1U8qSDq)

Comment (EdwardDowling) [reply integrations/access/jira/app.go]: Swapped to using the background context instead of the parent one.

**Comment by EdwardDowling** [comment]
Have confirmation the fix works [here](https://github.com/gravitational/customer-sensitive-requests/issues/304#issuecomment-2303186251)

**Review by r0mant — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**